test/stdlib/Inputs/CommonArrayTests.gyb

```diff
@@ -156,19 +156,17 @@ ${Suite}.test("${ArrayType}/filter") {
     let arr: ${ArrayType}<Int> = [ 0, 30, 10, 90 ]
     let result = arr.filter() { (x: Int) -> Bool in true }
     expectEqual([ 0, 30, 10, 90 ], result)
-    expectGE(2 * result.count, result.capacity)
```

Comment:
Was dropping these checks intentional? It's not obvious to me why Substring having its own views would affect Array.